src/sortStudents.ts

```diff
@@ -1,16 +1,60 @@
+/* eslint-disable max-len */
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: true,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function calculateAverage(grades: number[]): number {
+  return grades.reduce((total, grade) => total + grade, 0) / grades.length;
+}
+
+export function sortStudents(students: Student[], sortBy: SortType, order: SortOrder): Student[] {
+  const copiedStudents = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+      copiedStudents.sort((a, b) => (order === 'asc' ? a.name.localeCompare(b.name) : b.name.localeCompare(a.name)));
+      break;
+    case SortType.Surname:
+      copiedStudents.sort((a, b) => (order === 'asc' ? a.surname.localeCompare(b.surname) : b.surname.localeCompare(a.surname)));
```

Comment:
its code duplicating, if you will describe your enum you can use here smth like this
```suggestion
    case SortType.Name:
    case SortType.Surname:
      copiedStudents.sort((a, b) => (
        order === 'asc' 
          ? a[sortBy].localeCompare(b[sortBy]) 
          : b[sortBy].localeCompare(a[sortBy])));
      break;
```
and use better naming for your sort args like, first second, etc.